### PR TITLE
feat: social media share button on pfp borrower profiles

### PR DESCRIPTION
**Pull request by eddieferrer**
* refactor some shared settings code, remove lodash
* upgrade kv elements to get new button variant
* add button on loan cta to open modal and share on social media

ACK-303

Button is only displayed for PFP loans. I am ignoring the share language experiment for this use case. 

Uses a slot because button is positioned differently based on screen resolution. Screenshots:
Screen Shot 2022-07-20 at 1 58 58 PM
Screen Shot 2022-07-20 at 1 58 32 PM
Screen Shot 2022-07-20 at 1 58 23 PM
Screen Shot 2022-07-20 at 12 59 39 PM

### CODE REVIEW

File: tailwind.purge.safelist.txt [hunk 1]
```diff
@@ -81,6 +81,9 @@ tw-text-link
 tw-whitespace-normal
 tw-whitespace-pre-wrap
 
+# Additional Grid Classes
```
Comment (eddieferrer) [inline tailwind.purge.safelist.txt:84]: I don't know why I have to whitelist this. I don't think our JIT is working quite right....

File: src/util/urlUtils.js [hunk 2]
```diff
@@ -38,3 +38,20 @@ export function isExcludedUrl(urlArray, route) {
...
 }
+
+/**
+ * Given a base url and query params in the form of object key value pair, return a fully formed url
+ *
+ * @param {string} base - The base url string
+ * @param {string} args - query params in the form of object key value pairs
+ * @returns {string}
+ */
+export function getFullUrl(base, args) {
```
Comment (eddieferrer) [inline src/util/urlUtils.js:49]: Moved this to util function out of components

File: src/components/BorrowerProfile/SocialShareButton.vue [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+.social-button {
+	@apply tw-mb-2.5 tw-w-fit;
+}
+
+.social-button__icon {
+	@apply tw-w-5 tw-h-5 tw-pointer-events-none tw-inline-block tw-align-middle;
+}
+</style>
+
+<style lang="scss" scoped>
```
Comment (dyersituations) [inline src/components/BorrowerProfile/SocialShareButton.vue:247]: Do we want to replace the `scss` with `postcss`, or are we still using SCSS in the `ui` project?

**Review by dyersituations — APPROVED** [review]

File: src/components/BorrowerProfile/SocialShareButton.vue (hunk 3 above)
Comment (mcstover) [reply src/components/BorrowerProfile/SocialShareButton.vue:247]: It's worth having a look to see if postcss will cover all of the existing usages of scss across the Ui repo...

Comment (eddieferrer) [reply src/components/BorrowerProfile/SocialShareButton.vue:247]: using `lang="postcss"` will allow you to @apply tailwind classes. It does NOT allow nested CSS, while `lang="scss"` allows nesting and variables, but it does not allow you to @apply tailwind classes

Comment (mcstover) [reply src/components/BorrowerProfile/SocialShareButton.vue:247]: Right right, cheers to multiple style blocks!

**Review by mcstover — APPROVED** [review]

File: src/pages/BorrowerProfile/BorrowerProfile.vue [hunk 4]
```diff
@@ -21,16 +21,21 @@
...
-						class="tw-relative lg:tw--mb-1.5 tw-z-1"
+						class="tw-relative lg:tw--mb-1.5"
 						:show-urgency-exp="showUrgencyExp"
 						:lenders="lenders"
 						:social-exp-enabled="socialExpEnabled"
-					/>
+					>
+						<template #sharebutton v-if="inPfp">
+							<!-- Share button for PFP loans -->
+							<social-share-button class="tw-block md:tw-hidden tw-mt-3" :loan="loan" :lender="lender" />
```
Comment (mcstover) [inline src/pages/BorrowerProfile/BorrowerProfile.vue:31]: Is this "first version" of sharing on BP tightly coupled to the `inPfp` state or could we make it available to the general public as well...

Comment (eddieferrer) [reply src/pages/BorrowerProfile/BorrowerProfile.vue:31]: The spec was just for PFP profiles. I will bring this up to the rest of the team. I see no reason why this couldn't be opened up to all profiles.

**Pull request merged**